cmd/dlt/machinepool/cmd.go

```diff
@@ -53,13 +54,16 @@ func run(_ *cobra.Command, argv []string) {
 	r := rosa.NewRuntime().WithAWS().WithOCM()
 	defer r.Cleanup()
 
-	machinePoolID := argv[0]
+	machinePoolId := argv[0]
 	clusterKey := r.GetClusterKey()
 	cluster := r.FetchCluster()
 
-	if cluster.Hypershift().Enabled() {
-		deleteNodePool(r, machinePoolID, clusterKey, cluster)
-	} else {
-		deleteMachinePool(r, machinePoolID, clusterKey, cluster)
+	service := machinepool.NewMachinePoolService()
+	if !machinepool.MachinePoolKeyRE.MatchString(machinePoolId) {
```

Comment:
it‘s better to move the machineId validation right after we got its' value "machinePoolId := argv[0]"